haskell/test/Test/Ligo/Common.hs

```diff
@@ -45,10 +47,29 @@ class VariantConstraints variant =>
   toProposalMetadata = toMetadata @variant
   getVariantStorageRPC :: forall caps m . MonadCleveland caps m => ContractAddress -> m (StorageSkeletonRPC (VariantToExtra variant))
 
+withOriginatedFA12
+  :: forall variant num num' caps m a num0. (IsoNatPeano num num', SingI num', TestableVariant variant, MonadCleveland caps m, num' ~ 'S num0, MonadFail m)
+  => (SizedList num ImplicitAddress -> StorageSkeleton (VariantToExtra variant) -> StorageSkeleton (VariantToExtra variant))
+  -> (SizedList num ImplicitAddress
+        -> StorageSkeleton (VariantToExtra variant)
+        -> ContractHandle (Parameter' (VariantToParam variant)) (StorageSkeleton (VariantToExtra variant)) ()
+        -> ContractHandle FA2.Parameter [FA2.TransferParams] ()
+        -> ContractHandle FA12.Parameter [FA12.TransferParams] ()
+        -> m a)
+  -> m a
+withOriginatedFA12 storageFn tests = do
+  withOriginated @variant storageFn (\addrs s tc fa2 -> do
+    dodFA12TokenContract <- originate "token_contract" [] dummyFA12Contract
```

Comment:
Tests take a long time, and timings are annoying as is, let's not exacerbate the issue.

Please refactor `withOriginated` to accept a parameter which type of token it expects instead of this horrible hack. I'm objecting to this on the grounds that FA1.2 contract is originated during the initial voting period, which will throw off timings, and those are super annoying to deal with as it is.

Reply:
> I'm objecting to this on the grounds that FA1.2 contract is originated during the initial voting period, which will throw off timings

Good catch. I have tried a fix in https://github.com/tezos-commons/baseDAO/pull/345/commits/52b8dfcaf174b63bfdc19afc7e9170608886eb4f